app/controllers/carts_controller.rb

```diff
@@ -1,25 +1,40 @@
 class CartsController < ApplicationController
   before_action :set_cart, only: [:show,:create,:destroy]
+
+
+
   def show
-    @cart_items= CartItem.joins("INNER JOIN items ON cart_items.item_id=items.id").where(cart_id:@cart.id).pluck(:item_id,:title,:quantity)
+    @cart_items= CartItem.get_cart_items(@cart)
   end
+
+
   #create function here do not create new cart but just add item to cart
   def create
     @item=Item.find_by(id:params[:item_id])
 
-    if @item
+    if @item && @cart
 
       @cart.add_item(@item)
       redirect_to items_path
+    else
+      redirect_to record_not_found_home_path(1)
```

Comment:
Why are passing 1 here. And If it is a parameter is should be with some name

Reply:
removed 1